meow/user_profile/views.py

```diff
@@ -5,29 +5,146 @@
 from django.core import serializers
 
 from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from meow.utils.decorators import api_login_required
+
 from user_profile.models import User, Theme
 from user_profile.serializers import SafeUserSerializer, ThemeSerializer
 import json
 # Create your views here.
 
+class UserThemes(APIView):
+    def get(self, request, id, format=None):
+        if not request.user.is_authenticated:
+            return HttpResponse('Must be logged in', status=403)
+        user = request.user
+        themes = Theme.objects.filter(name="Daily Bruin")| Theme.objects.filter(name="Dark Bruin") | Theme.objects.filter(author=user)
+        serialized_themes = ThemeSerializer(themes, many=True)
+        themeOrderedDict = serialized_themes.data
+        return JsonResponse(themeOrderedDict, safe=False, status=200)
+    
+    def post(self, request, id, format=None):
+        if not request.user.is_authenticated:
+            return HttpResponse('Must be logged in', status=403)
+        user = request.user
+        themes = Theme.objects.all()
+        req_data = request.data
+        new_name = req_data.get("name", None)
+        new_primary = req_data.get("primary", None)
+        new_secondary = req_data.get("secondary", None)
+        new_primary_font_color = req_data.get("primary_font_color", None)
+        new_secondary_font_color = req_data.get("secondary_font_color", None)
+        new_tertiary = req_data.get("tertiary", None)
+        if new_name == "":
+            return JsonResponse('Theme name cannot be empty', safe=False, status=400)
+        if (themes.filter(name=new_name)):
```

Comment:
i dont really know how to use serializers because author in this case is an object

Reply:
i don't really know how to use serializers because the author of each theme is an object so they are represented by ids on the client side

Reply:
this is because on the models side the author field is a foreign key field but when i send the stuff over from the frontend, i cant send the author over, in fact the author send over are just numbers, how do i work with that?

Reply:
Ahhh I c. So thankfully django rest framework has something for this!
https://www.django-rest-framework.org/api-guide/relations/#primarykeyrelatedfield
primarykeyrelatedfield. If you do something like
```
class ThemeSerializer(..):
    author = PrimaryKeyRelatedField()
  
    class Meta:
           model = Theme
           fields = ("author", "name", ...
```

So when the backend serializes a Theme object, it will be something like this 
```
{
    "author": 3,
   "name": "my theme",
    . . .
}
```

Reply:
what does the primary key related field do, it take is an id and find the corresponding object and vice-versa??

Reply:
also do i have to define create and save methods in the serializer